.github/workflows/UploadDockerImage.yml

```diff
@@ -0,0 +1,44 @@
+# Pushes Docker images created from the deploy tool's Dockerfile templates to an internal ECR so that they can be scanned for security vulnerabilities.
+name: Upload Docker Images
+
+on:
+  # Manually trigger on specific branches
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+
+permissions:
+  id-token: write
+
+jobs:
+  upload-docker-images:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@8c3f20df09ac63af7b3ae3d7c91f105f857d8497 #v4
```

Comment:
why not specify `aws-actions/configure-aws-credentials@v4` instead of specifiying the commit? I prefer we remain consistent with how we currently do this.

Reply:
I had pinned this GitHub action based on it version number but it was flagged by CodeQL - https://github.com/aws/aws-dotnet-deploy/security/code-scanning/18. So I changed it to use the full commit SHA.